### DISCUSSION TITLE
Is there a BNF description for Vega-Lite?

### DISCUSSION THREAD

**Discussion by jw5** [vega/vega-lite#8481] [Q&A]
While the on-line vega-lite is very useful, and often there are authoritative textual descriptions of what is allowed, there are occasions where I just can't tell what an acceptable value is within a given statement. I was hoping that there was a formal BNF (like) description of permitted input. Since the on-line editor does a fair (but not perfect) job of identifying violations I thought perhaps there was a general specification already available?

Thanks,
Jim

**Comment by domoritz** [comment]
You can use the json schema we provide for both Vega-Lite and Vega. In fact, it's what powers autocomplete in the vega editor. Alternatively, you can use the typescript types that are also complete. We had a partial bnf in the Vega-Lite paper but it's of course outdated.
